spec/request/api/v1/auth/reset_password_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+
+describe 'PUT api/v1/auth/password', type: :request do
+  let(:user) { create(:user, :confirmed) }
+  let(:reset_url) { email_link_url last_email }
+  let(:params) do
+    {
+      email: user.email,
+      redirect_url: 'http://localhost:3000'
+    }
+  end
+  let(:new_user) { build(:user) }
+  let(:new_params) do
+    {
+      password: new_user.password,
+      password_confirmation: new_user.password
+    }
+  end
+
+  before do
+    post user_password_path params
+  end
+
+  context 'when the request is succesful' do
+    before do
+      get reset_url
+      headers = url_query_parameters(response.location)
+      put user_password_path, params: new_params, headers: headers
+    end
+
+    it 'is expected a successful response' do
+      expect(response).to be_successful
+    end
+
+    it 'is expected that response contains some body data' do
+      body = JSON response.body
+      expect(json_value(body, 'data', 'id')).not_to be_nil
+      expect(json_value(body, 'data', 'uid')).to eq user.email
+      expect(json_value(body, 'data', 'email')).to eq user.email
+      expect(json_value(body, 'data', 'name')).to eq user.name
+    end
+
+    it 'is expected a successful login with new password' do
```

Comment:
You could just try if the password has actually changed with: `user.valid_password? new_password`